bokeh/server/application_context.py

```diff
@@ -15,8 +15,9 @@ class ApplicationContext(object):
         data specific to an "instance" of the application.
     '''
 
-    def __init__(self, application):
+    def __init__(self, application, io_loop=None):
```

Comment:
not sure. My understanding is that application_context should just take an io_loop that is being created somewhere else (an "application manager" let's say, like `BokehTornado`, that lives on the server that is creating the io_loop). Does the `ApplicationContext` makes sense outside of the `server` context at all?

If not  then maybe it'd actually make sense to turn io_loop into a positional argument and expect it to always be specified..